src/library_webgpu.js

```diff
@@ -2575,16 +2575,20 @@ var LibraryWebGPU = {
     assert({{{ gpu.PresentMode.Fifo }}} ===
       {{{ gpu.makeGetU32('descriptor', C_STRUCTS.WGPUSwapChainDescriptor.presentMode) }}});
 #endif
+    var configurationSize = [
+      {{{ gpu.makeGetU32('descriptor', C_STRUCTS.WGPUSwapChainDescriptor.width) }}},
+      {{{ gpu.makeGetU32('descriptor', C_STRUCTS.WGPUSwapChainDescriptor.height) }}},
+    ];
 
     var configuration = {
       "device": device,
       "format": WebGPU.TextureFormat[
         {{{ gpu.makeGetU32('descriptor', C_STRUCTS.WGPUSwapChainDescriptor.format) }}}],
       "usage": {{{ gpu.makeGetU32('descriptor', C_STRUCTS.WGPUSwapChainDescriptor.usage) }}},
-      "size": [
-        {{{ gpu.makeGetU32('descriptor', C_STRUCTS.WGPUSwapChainDescriptor.width) }}},
-        {{{ gpu.makeGetU32('descriptor', C_STRUCTS.WGPUSwapChainDescriptor.height) }}},
-      ],
+      "size": configurationSize[0] !== 0 && configurationSize[1] !== 0
+        ? configurationSize
+        : undefined,
+        "alphaMode": "opaque",
```

Comment:
When the configurationSize is *not* zero, I think we should set the size of the canvas (`context["canvas"]["width"]` etc), because otherwise (at least at the moment) we'll be changing the behavior compared to how this C API is supposed to work on native. That API may change in the future but we should adhere to it for now.

Reply:
That is to say - instead of specifying the "size" member in that case, which is deprecated, I think we should actually implement it properly by setting the canvas size attributes.

Reply:
I can change it, but is there a way to get the current size of the canvas on the C++ side? If there's not would there be a way to set the swapchain to just take the current size of the canvas?

Reply:
It should be all good now, let me know if there's something else you would change

Reply:
> I can change it, but is there a way to get the current size of the canvas on the C++ side?

Yes, you can use [`emscripten_get_canvas_element_size`](https://emscripten.org/docs/api_reference/html5.h.html#c.emscripten_get_canvas_element_size) if needed.

> If there's not would there be a way to set the swapchain to just take the current size of the canvas?

Not sure I understand. Passing 0,0 of course skips setting `canvas.width`/`canvas.height`. And `configure()` will no longer change the size since that was the deprecated behavior.